### DISCUSSION TITLE
Expander: Conditional Expand?

### DISCUSSION THREAD

**Discussion by ajsuydam** [CommunityToolkit/Maui#1995] [Q&A]
After reading the docs, I am still wondering just a bit on a question: is it possible to only expand the content of the expander if there is some property set that allows that? For instance, if I have an object that may or may not have some field, I want to only expand the details of that object if it contains that field (i.e. the field is != null).

Appreciate any thoughts anyone might have.

**Comment by ghost** [comment]
You would set this in your controller, for example IsExpanded = false. Then on get you could check (field is != null) ? true : false. So isExpanded in XAML, would be true/false based on your field value. 

Just make sure when you update your "field" value, you must call OnPropertyChanging(nameof(IsExpanded)) and OnPropertyChanged(nameof(IsExpanded)).

So when "field" value will change also the Bindable IsExpanded in XAML will be notified that has changed and expander UI will update.
